elastic/kibana
# [Security Solution][Alerts] Pass filters from threshold alerts to timeline

## Summary
Rule filters stored on threshold alerts were not being passed to timeline. This PR passes the filters through. That exposes https://github.com/elastic/kibana/issues/126574 (filters without `meta` cause a stack trace in the UI) in `main`, which was previously only occurring in the 7.x series. This PR fixes that issue as well by providing a default value for `meta` if it doesn't exist.

## Details
`createThresholdTimeline` was expecting serialized filters as an array of strings and attempting to deserialize them before passing to `createTimeline`. However, the filters in `formattedAlertData` are already deserialized. As a result, `getFiltersFromRule` was returning an empty array instead of the actual filters. This was causing the filters from threshold rules to not be included in the timeline view.

The fix is simply to remove `getFiltersFromRule` since we don't need to deserialize the filters anymore. Unit tests updated to verify that the expected filters are passed to `createTimeline`, and in the UI filters are now correctly included in the timeline after clicking the "Investigate in timeline" button on an alert (screenshot below).

![image](https://user-images.githubusercontent.com/55718608/161622152-962170fb-dcfd-464a-884e-ce45a44089d5.png)

## Testing
1. Create documents to trigger the rule. Adjust the `@timestamp` and `event.ingested` values to current time and make a second document with a different `source.geo.country_name` value.
```
PUT auditbeat-1
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      },
      "event.category": {
        "type": "keyword"
      },
      "event.ingested": {
        "type": "date"
      },
      "event.outcome": {
        "type": "keyword"
      },
      "source.ip": {
        "type": "keyword"
      },
      "source.geo.country_name": {
        "type": "keyword"
      },
      "user.name": {
        "type": "keyword"
      },
      "host.name": {
        "type": "keyword"
      },
      "otherField": {
        "type": "keyword"
      }
    }
  }
}

POST auditbeat-1/_doc
{
  "@timestamp": 1649109892000,
  "event.ingested": 1649109892000,
  "user.name": "you",
  "event.category": "authentication",
  "event.outcome": "success",
  "source.ip": "test",
  "source.geo.country_name": "usa"
}
```
2. Create a threshold rule with a filter that doesn't have a `meta` field. Easiest way is to import the rules file [here](https://gist.github.com/marshallmain/0c24f772c9be212b29da423ddbc7c999). The rule will run on import and should create an alert.
3. Click `Investigate in timeline` on the alert. It should successfully open the timeline and include the filter defined on the rule.

Related: https://github.com/elastic/kibana/issues/126574
7.17: https://github.com/elastic/kibana/pull/129427
